cmd/multiclusterapp.go

```diff
@@ -0,0 +1,703 @@
+	}
+
+	if ctx.NArg() < 2 {
+		return cli.ShowSubcommandHelp(ctx)
+	}
+
+	resource, err := Lookup(c, ctx.Args().First(), managementClient.MultiClusterAppType)
+	if err != nil {
+		return err
+	}
+
+	app, err := c.ManagementClient.MultiClusterApp.ByID(resource.ID)
+	if err != nil {
+		return err
+	}
+
+	answers := fromMultiClusterAppAnswers(app.Answers)
+	err = processAnswers(ctx, c, nil, answers, false)
+	if err != nil {
+		return err
+	}
+	app.Answers = toMultiClusterAppAnswers(answers)
+
+	newTargets := ctx.StringSlice("target")
+	if len(newTargets) > 0 {
+		app.Targets = nil
+		for _, target := range newTargets {
+			app.Targets = append(app.Targets, managementClient.Target{
+				ProjectID: target,
+			})
+		}
+	}
+
+	_, err = c.ManagementClient.MultiClusterApp.Update(app, app)
+	return err
+}
+
+func multiClusterAppRollback(ctx *cli.Context) error {
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	if ctx.Bool("show-revisions") {
+		return outputMultiClusterAppRevisions(ctx, c)
+	}
+
+	if ctx.NArg() < 2 {
+		return cli.ShowSubcommandHelp(ctx)
+	}
+
+	resource, err := Lookup(c, ctx.Args().First(), managementClient.MultiClusterAppType)
+	if err != nil {
+		return err
+	}
+
+	app, err := c.ManagementClient.MultiClusterApp.ByID(resource.ID)
+	if err != nil {
+		return err
+	}
+
+	revisionResource, err := Lookup(c, ctx.Args().Get(1), managementClient.MultiClusterAppRevisionType)
+	if err != nil {
+		return err
+	}
+
+	rr := &managementClient.MultiClusterAppRollbackInput{
+		RevisionID: revisionResource.ID,
+	}
+	err = c.ManagementClient.MultiClusterApp.ActionRollback(app, rr)
+	return err
+}
+
+func multiClusterAppTemplateInstall(ctx *cli.Context) error {
+	if ctx.NArg() == 0 {
+		return cli.ShowSubcommandHelp(ctx)
+	}
+	templateName := ctx.Args().First()
+	appName := ctx.Args().Get(1)
+
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	app := &managementClient.MultiClusterApp{
+		Name: appName,
+	}
+
+	resource, err := Lookup(c, templateName, managementClient.TemplateType)
+	if err != nil {
+		return err
+	}
+	template, err := c.ManagementClient.Template.ByID(resource.ID)
+
+	templateVersionID := templateVersionIDFromVersionLink(template.VersionLinks[template.DefaultVersion])
+	userVersion := ctx.String("version")
+	if userVersion != "" {
+		if link, ok := template.VersionLinks[userVersion]; ok {
+			templateVersionID = templateVersionIDFromVersionLink(link)
+		} else {
+			return fmt.Errorf(
+				"version %s for template %s is invalid, run 'rancher mcapp show-template %s' for a list of versions",
+				userVersion,
+				templateName,
+				templateName,
+			)
+		}
+	}
+
+	templateVersion, err := c.ManagementClient.TemplateVersion.ByID(templateVersionID)
+	if err != nil {
+		return err
+	}
+
+	interactive := !ctx.Bool("no-prompt")
+	answers := make(map[string]string)
+	err = processAnswers(ctx, c, templateVersion, answers, interactive)
+	if err != nil {
+		return err
+	}
+
+	targets := ctx.StringSlice("target")
```

Comment:
Lookup can be used to get projects by name, this would be consistent with other resources and also the UI

Reply:
Updated